velox/functions/sparksql/tests/ArrayInsertTest.cpp

```diff
@@ -0,0 +1,260 @@
+  testExpression(
+      "array_insert(c0, 1, null::BOOLEAN, false)", {arrays}, expected);
+
+  expected = expected = makeArrayVectorFromJson<bool>(
+      {"[true, true]", "[true, true, false]", "[true, false, true, true]"});
+  testExpression("array_insert(c0, -2, true, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, strArrays) {
+  using S = StringView;
+
+  auto arrays = makeNullableArrayVector<StringView>({
+      {S("a"), std::nullopt, S("b")},
+      {S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, S("b"), std::nullopt},
+      {S("purple is an elegant color")},
+  });
+
+  auto expected = makeNullableArrayVector<StringView>({
+      {S("X"), S("a"), std::nullopt, S("b")},
+      {S("X"), S("a"), S("b"), S("a"), S("a")},
+      {S("X"), std::nullopt, S("b"), std::nullopt},
+      {S("X"), S("purple is an elegant color")},
+  });
+  testExpression("array_insert(c0, 1, 'X', false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>({
+      {std::nullopt, S("a"), std::nullopt, S("b")},
+      {std::nullopt, S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, std::nullopt, S("b"), std::nullopt},
+      {std::nullopt, S("purple is an elegant color")},
+  });
+  testExpression(
+      "array_insert(c0, 1, null::STRING, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>({
+      {S("a"), std::nullopt, S("X"), S("b")},
+      {S("a"), S("b"), S("a"), S("X"), S("a")},
+      {std::nullopt, S("b"), S("X"), std::nullopt},
+      {S("X"), S("purple is an elegant color")},
+  });
+  testExpression("array_insert(c0, -2, 'X', false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, varbinary) {
+  auto arrays = makeNullableArrayVector<StringView>(
+      {{"a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+
+  auto expected = makeNullableArrayVector<StringView>(
+      {{"X"_sv, "a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, 1, 'X'::VARBINARY, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>(
+      {{std::nullopt, "a"_sv, "b"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, 1, null::VARBINARY, false)", {arrays}, expected);
+
+  expected = makeNullableArrayVector<StringView>(
+      {{"a"_sv, "b"_sv, "X"_sv, "c"_sv}}, ARRAY(VARBINARY()));
+  testExpression(
+      "array_insert(c0, -2, 'X'::VARBINARY, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, nullSrcArrays) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"null"});
+  const auto expected = makeArrayVectorFromJson<int64_t>({"null"});
+  testExpression("array_insert(c0, 1, 1, false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, nullPosition) {
+  const auto arrays = makeArrayVectorFromJson<int64_t>({"[1, 1]"});
+  const auto expected = makeArrayVectorFromJson<int64_t>({"null"});
+  testExpression(
+      "array_insert(c0, null::INTEGER, 1, false)", {arrays}, expected);
+}
```

Comment:
Perhaps combine nullSrcArrays and nullPosition as nullInputs. Also add a case for the new element being to verify that the result is not null.

Reply:
Done.